sycl/include/CL/sycl/INTEL/ac_int.hpp

```diff
@@ -0,0 +1,2458 @@
+    // use a customized version of strlen instead
+    const char *s = nullptr;
+    for (s = str; *s; ++s) {
+    }
+    int str_len = (s - str);
+    ac_int<W, S> res = 0;
+    if (str_len >= 2 && str[0] == '0' && str[1] == 'x') {
+#pragma unroll
+      // the first two digits can be prefix "0x"
+      for (int i = 2; i < str_len; i++) {
+        char c = str[i];
+        ac_int<4, false> h = 0;
+        if (c >= '0' && c <= '9')
+          h = c - '0';
+        else if (c >= 'A' && c <= 'F')
+          h = c - 'A' + 10;
+        else if (c >= 'a' && c <= 'f')
+          h = c - 'a' + 10;
+        else {
+          AC_ASSERT(!c, "Invalid hex digit");
+          break;
+        }
+        ac_int<4, false> s = 4;
+        res = res << s;
+        res |= h;
+      }
+      *this = res;
+    } else {
+#pragma unroll
+      // the first two digits can be prefix "0x"
+      for (int i = 0; i < str_len; i++) {
+        char c = str[i];
+        ac_int<4, false> h = 0;
+        if (c >= '0' && c <= '9')
+          h = c - '0';
+        else if (c >= 'A' && c <= 'F')
+          h = c - 'A' + 10;
+        else if (c >= 'a' && c <= 'f')
+          h = c - 'a' + 10;
+        else {
+          AC_ASSERT(!c, "Invalid hex digit");
+          break;
+        }
+        ac_int<4, false> s = 4;
+        res = res << s;
+        res |= h;
+      }
+      *this = res;
+    }
+  }
+
+  template <int Na>
+  constexpr void bit_fill(const int (&ivec)[Na], bool bigendian = true) {
+    // bit_fill from integer vector
+    //   if W > N*32, missing most significant bits are zeroed
+    //   if W < N*32, additional bits in ivec are ignored (no overflow checking)
+    // Example:
+    //   ac_int<80,false> x;
+    //   int vec[] = {
+    //     0xffffa987, 0x6543210f, 0xedcba987
+    //   };
+    //   x.bit_fill(vec);   // vec[0] fill bits 79-64
+    const int M = AC_MIN((W + 31) / 32, Na);
+    ac_int<M * 32, false> res = 0;
+#pragma unroll
+    for (int i = 0; i < M; i++) {
+      res.set_slc(i * 32, ac_int<32, false>(ivec[bigendian ? M - 1 - i : i]));
+    }
+    *this = res;
+  }
+
+  void _set_value_internal(typename Base::actype _value) {
+    this->value = _value;
+  }
+  const typename Base::actype _get_value_internal() const {
+    return this->value;
+  }
+};
+
+namespace ac {
+template <typename T, typename T2> struct rt_2T {
+  typedef typename ac_private::map<T>::t map_T;
+  typedef typename ac_private::map<T2>::t map_T2;
+  typedef typename map_T::template rt_T<map_T2>::mult mult;
+  typedef typename map_T::template rt_T<map_T2>::plus plus;
+  typedef typename map_T::template rt_T<map_T2>::minus minus;
+  typedef typename map_T::template rt_T<map_T2>::minus2 minus2;
+  typedef typename map_T::template rt_T<map_T2>::logic logic;
+  typedef typename map_T::template rt_T<map_T2>::div div;
+  typedef typename map_T::template rt_T<map_T2>::div2 div2;
+};
+} // namespace ac
+
+namespace ac {
```

Comment:
```suggestion
 
```

Reply:
This might make the header incompatible with HLS compiler hence I will not make this change initially. Let me know if you object strongly.

Reply:
Okay, I don't mind